### PR TITLE
[Enhancement] Optimize task query metadata when there are many dbs

### LINKED ISSUE #6892: Task query INFORMATION_SCHEMA may be slow when there are many dbs
## Enhancement
his is because each query has to send a number of RPC requests to the DB, we can try to optimize it.

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6892

## Problem Summary(Required) ：

We can first send an RPC to count which DBs all Task/TaskRun contains, and then reduce the RPC query for each DB.
Usually, users do not build many DBs and send Tasks in different DBs at the same time.
It can be further optimized later, and the indexes of which DBs are included in the Task are built in advance. At present, no deep optimization is performed.

### CODE REVIEW

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -294,6 +294,7 @@ struct TGetDbsParams {
   2: optional string user   // deprecated
   3: optional string user_ip    // deprecated
   4: optional Types.TUserIdentity current_user_ident // to replace the user and user ip
+  5: optional string filter_type
```
Comment (imay) [inline gensrc/thrift/FrontendService.thrift:297]: Could you give some comment for this field.
Why not use enum for this type?

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Review by imay — COMMENTED** [review]
Why you should get db first? I think you should just get task and task runs directly.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|0|3|00.00%|

**Comment by Astralidea** [comment]
> Why you should get db first? I think you should just get task and task runs directly.

I think If the request is made at the db granularity, the permission check only needs to be done once for one db. If the request is not made at the db granularity, the permission check needs to be check for every Task/TaskRun.
However, it is possible that this check has little impact on memory, and you prefer to get all Tasks directly at once?

File: gensrc/thrift/FrontendService.thrift (hunk 1 above)
Comment (Astralidea) [reply gensrc/thrift/FrontendService.thrift:297]: I will remove this field and try get all task/taskrun once.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|

**Review by ABingHuang — COMMENTED** [review]
LGTM

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|

File: be/src/exec/vectorized/schema_scanner/schema_tasks_scanner.h [hunk 2]
```diff
@@ -16,10 +16,8 @@ class SchemaTasksScanner : public SchemaScanner {
     Status get_next(ChunkPtr* chunk, bool* eos) override;
 
 private:
-    Status get_new_tasks();
     Status fill_chunk(ChunkPtr* chunk);
 
-    int _db_index{0};
     int _task_index{0};
     TGetDbsResult _db_result;
```
Comment (wyb) [inline be/src/exec/vectorized/schema_scanner/schema_tasks_scanner.h:22]: `_db_result` is not used.

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java [hunk 3]
```diff
@@ -371,19 +372,22 @@ public TGetTaskRunInfoResult getTaskRuns(TGetTasksParams params) throws TExcepti
...
+        TaskManager taskManager = globalStateMgr.getTaskManager();
+        List<TaskRunStatus> taskRunList = taskManager.showTaskRunStatus(null);
 
-        TaskManager taskManager = GlobalStateMgr.getCurrentState().getTaskManager();
-        List<TaskRunStatus> taskRunList = taskManager.showTaskRunStatus(params.db);
         for (TaskRunStatus status : taskRunList) {
+
+            Database db = globalStateMgr.getDb(status.getDbName());
+            if (!globalStateMgr.getAuth().checkDbPriv(currentUser, db.getFullName(), PrivPredicate.SHOW)) {
+                return result;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:388]: Why return, not continue?

Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java]: mistake

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|

**Review by ABingHuang — APPROVED** [review]
LGTM

**Review by chaoyli — APPROVED** [review]

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.3

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|

**Pull request merged**